### PR TITLE
Makefile: TBB is a dependency of bin helpers

### PR DESCRIPTION
**Pull request by WardBrian**
#### Submisison Checklist

- [x] Run tests: `./runCmdStanTests.py src/test`
- [x] Declare copyright holder and open-source license: see below

#### Summary:

Currently, running `make bin/stansummary` without first running `make build` fails, because these link TBB but do not have it marked as a dependency.

This can, rarely, cause `make build` to fail, like in CI here: https://jenkins.flatironinstitute.org/blue/organizations/jenkins/Stan%2FStan/detail/downstream_tests/366/pipeline

#### Intended Effect:

Fix `make bin/stansummary`

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company):
Simons Foundation

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

File: make/command [hunk 1]
```diff
@@ -2,13 +2,13 @@ ifeq ($(CMDSTAN_SUBMODULES),1)
 bin/cmdstan/stansummary.o : src/cmdstan/stansummary_helper.hpp
 bin/cmdstan/%.o : src/cmdstan/%.cpp
 	@mkdir -p $(dir $@)
-	$(COMPILE.cpp) -fvisibility=hidden $(OUTPUT_OPTION) $(LDLIBS) $<
+	$(COMPILE.cpp) -fvisibility=hidden $< $(OUTPUT_OPTION)
```
Comment (syclik) [inline make/command:5]: Was this change intended? I think it's a good change: the use of libs shouldn't be necessary when compiling, only linking.

Comment (WardBrian) [reply make/command:5]: Yes, it currently generates warnings for unused arguments, so I removed it

**Review by syclik — APPROVED** [review]

**Comment by syclik** [comment]
Awesome. Approved.
